flexget/plugins/urlrewrite_newpct.py

```diff
@@ -20,27 +29,56 @@ class UrlRewriteNewPCT(object):
 
     # urlrewriter API
     def url_rewritable(self, task, entry):
-        url = entry['url']
-        rewritable_regex='^http:\/\/(www.)?newpct1?.com\/.*'
-        return re.match(rewritable_regex,url) and not url.startswith('http://www.newpct.com/descargar/')
+        if entry.get('urls'):
+            url = entry.get('urls')[0]
+        else:
+            url = entry['url']
+        rewritable_regex = '^http:\/\/(www.)?newpct1?.com\/.*'
+        return (re.match(rewritable_regex, url) and not url.startswith('http://www.newpct.com/descargar/') and not
+                url.endswith('.torrent') and not url.startswith('http://www.newpct1.com/descargar-torrent/'))
 
     # urlrewriter API
     def url_rewrite(self, task, entry):
-        entry['url'] = self.parse_download_page(entry['url'])
+        if 'newpct1' in entry['url']:
+            auth_handler = NewPCTAuth()
+            entry['download_auth'] = auth_handler
+            entry['urls'] = [self.parse_download_page_newpct1(task, entry['url'])]
+        else:
+            entry['urls'] = [self.parse_download_page(task, entry['url'])]
+        task.requests = Session() 
+    
+    def parse_download_page_newpct1(self, task, url):
+        log.debug('URL newpct1')
+        log.debug(url)
+        url = url.replace('newpct1.com/', 'newpct1.com/descarga-torrent/')
+        page = task.requests.get(url).content
+        soup = get_soup(page)
+        regex = re.compile(r'http:\/\/tumejorjuego\.com\/download\/index\.php\?link=descargar-torrent\/.+')
+        query = soup.findAll('a', href=regex)
+        if len(query) == 0:
+            raise UrlRewritingError('Unable to locate torrent from url %s' % url)
+        url = query[0]['href']
+        url = url.replace('http://tumejorjuego.com/download/index.php?link=', 'http://www.newpct1.com/')
+        return url
 
     @plugin.internet(log)
-    def parse_download_page(self, url):
-        page = requests.get(url)
+    def parse_download_page(self, task, url):
+        log.debug('URL newpct')
+        log.debug(url)
+        page = task.requests.get(url)
         try:
-            soup = get_soup(page.text)
+            soup = get_soup(page.content)
         except Exception as e:
             raise UrlRewritingError(e)
-        torrent_id_prog = re.compile("'(?:torrentID|id)'\s*:\s*'(\d+)'")
-        torrent_ids = soup.findAll(text=torrent_id_prog)
-        if len(torrent_ids) == 0:
-            raise UrlRewritingError('Unable to locate torrent ID from url %s' % url)
-        torrent_id = torrent_id_prog.search(torrent_ids[0]).group(1)
-        return 'http://www.newpct.com/descargar/torrent/%s/dummy.html' % torrent_id
+        torrent_span = soup.find('span', attrs={'id': 'content-torrent'})
+        try:
+            torrent_tag = torrent_span.findAll('a', href=True)
+        except Exception as e:
+            log.error('Error findAll')
+        if len(torrent_tag) == 0:
```

Comment:
If the above exception happens, `torrent_tag` will not be defined, and this will cause an exception.